File: angular-spring-ecommerce/backend/ecommerce-web/src/main/java/io/github/guilhermemelo01/ecommerceweb/dto/Purchase.java
package io.github.guilhermemelo01.ecommerceweb.dto;

import io.github.guilhermemelo01.ecommerceweb.entinty.Address;
import io.github.guilhermemelo01.ecommerceweb.entinty.Customer;
import io.github.guilhermemelo01.ecommerceweb.entinty.Order;
import io.github.guilhermemelo01.ecommerceweb.entinty.OrderItem;
import lombok.Data;

import java.util.HashSet;
import java.util.Set;

@Data
public class Purchase {

    private Customer customer;
    private Address billingAddress;
    private Address shippingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}

File: angular-spring-ecommerce/frontend/ecommerce-web/src/app/common/purchase.ts
import { OrderItem } from './order-item';
import { Order } from './order';
import { Address } from './address';
import { Customer } from './customer';

export class Purchase {

  customer: Customer;
  billingAddress: Address;
  shippingAdrress: Address;
  order: Order;
  orderItems: OrderItem[]
}
